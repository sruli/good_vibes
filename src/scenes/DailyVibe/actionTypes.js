export const CURRENT_IMAGE_CHANGED = 'DailyVibe.CURRENT_IMAGE_CHANGED';
export const PLAY_TAPPED = 'DailyVibe.PLAY_TAPPED';
export const PAUSE_TAPPED = 'DailyVibe.PAUSE_TAPPED';
export const SONG_FINISHED = 'DailyVibe.SONG_FINISHED';
export const TIMER_RESET = 'DailyVIbe.TIMER_RESET';
export const TIMER_SET = 'DailyVibe.TIMER_SET';
