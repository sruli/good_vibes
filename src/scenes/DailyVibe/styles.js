import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  backgroundImage: {
    width: null,
    height: null,
    flex: 1,
    justifyContent: 'flex-end',
  },
  controlBar: {
    marginBottom: '10%',
  },
});
