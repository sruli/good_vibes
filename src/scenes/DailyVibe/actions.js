import { CURRENT_IMAGE_CHANGED } from './actionTypes';

export const currentImageChanged = currentImage => ({
  type: CURRENT_IMAGE_CHANGED,
  payload: { currentImage },
});
