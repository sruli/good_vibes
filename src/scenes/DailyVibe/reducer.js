import { combineReducers } from 'redux';
import dataReducer from './data/reducer';

// actionTypes
import { VIBE_SETUP } from './data/vibe/actionTypes';
import {
  CURRENT_IMAGE_CHANGED,
  PLAY_TAPPED,
  PAUSE_TAPPED,
  TIMER_SET,
} from './actionTypes';

// selectors
import { getImages } from './data/vibe/reducer';

const playingReducer = (state = false, action) => {
  switch (action.type) {
    case PLAY_TAPPED:
      return true;
    case PAUSE_TAPPED:
      return false;
    default:
      return state;
  }
};

const currentImageReducer = (state = 0, action) => {
  switch (action.type) {
    case CURRENT_IMAGE_CHANGED:
      return action.payload.nextImage;
    default:
      return state;
  }
};

const timerKeyReducer = (state = null, action) => {
  switch (action.type) {
    case TIMER_SET:
      return action.payload.timerKey;
    default:
      return state;
  }
};

const musicPlayerReducer = (state = null, action) => {
  switch (action.type) {
    case VIBE_SETUP:
      return action.payload.musicPlayer;
    default:
      return state;
  }
};

export default combineReducers({
  playing: playingReducer,
  currentImage: currentImageReducer,
  timerKey: timerKeyReducer,
  musicPlayer: musicPlayerReducer,
  data: dataReducer,
});

export const getCurrentImage = state => (
  state.DailyVibe.currentImage
);

export const getPlaying = state => (
  state.DailyVibe.playing
);

export const getTimerKey = state => (
  state.DailyVibe.timerKey
);

export const getMusicPlayer = state => (
  state.DailyVibe.musicPlayer
);

// TODO test
export const getNextImage = (state) => {
  const currentImage = getCurrentImage(state);
  const images = getImages(state);
  const nextImage = currentImage + 1;

  return images[nextImage] ? nextImage : 0;
};

// TODO test
export const getPreviousImage = (state) => {
  const currentImage = getCurrentImage(state);
  const images = getImages(state);
  const previousImage = currentImage - 1;
  const lastIndex = images.length - 1;

  return previousImage >= 0 ? previousImage : lastIndex;
};
