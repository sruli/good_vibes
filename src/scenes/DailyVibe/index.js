import React from 'react';
import { Image, View } from 'react-native';
import styles from './styles';
import ControlBar from './components/ControlBar';

const DailyVibe = () => (
  // eslint-disable-next-line global-require
  <Image style={styles.backgroundImage} source={require('./assets/1.jpg')}>
    <View style={styles.controlBar}>
      <ControlBar />
    </View>
  </Image>
);

export default DailyVibe;
