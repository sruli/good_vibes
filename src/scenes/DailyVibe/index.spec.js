/* eslint-disable */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import DailyVibe from './index';

describe('<DailyVibe />', () => {
  it('renders a <DailyVibe>', () => {
    const tree = renderer.create(<DailyVibe />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
