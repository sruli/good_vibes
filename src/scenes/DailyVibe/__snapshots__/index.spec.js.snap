// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DailyVibe /> renders a <DailyVibe> 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": null,
      "justifyContent": "flex-end",
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": "10%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      />
    </View>
  </View>
</Image>
`;
